tests/zeromq.terminal/Program.cs

```diff
@@ -14,67 +13,68 @@ namespace zeromq.terminal
     ///</summary>
     class Program
     {
-        static void Main(string[] args)
+        static async Task Main(string[] args)
         {
             var version = args.FirstOrDefault();
             var configuration = SocketConfiguration.InprocConfig("this-inproc-sir-" + Guid.NewGuid());
             
             if (version != null)
-                RunPubSub(configuration);
+                await RunPubSub(configuration);
             else
-                RunReqRep(configuration);
+                await RunReqRep(configuration);
 
             NetMQConfig.Cleanup();
         }
 
-        private static void RunPubSub(SocketConfiguration configuration)
+        private static async Task RunPubSub(SocketConfiguration configuration)
         {
-            // only works with InProc
+            // it works only for InProc right now
             var pubSub = new PubSub(configuration);
```

Comment:
Since `PubSub` is `IDisposable`, shouldn't we call `PubSub.Dispose()` in this method instead - or have a C# 8.0 _using declaration_ instead, like `using var pubSub = new PubSub(configuration);`
Could we then get rid of `NetMQConfig.Cleanup();` in `Main(string[])`?
Or do I misunderstand here something?
...
Now that I think about it ... would it be a problem if I have multiple `PubSub` instances in use and then dispose just one of them, which calls the static `NetMQConfig.Cleanup();`? - Does it even make sense to have multiple `PubSub`s?
I guess there is something I didn't quite understand yet.

Reply:
No NetMqConfig.CleanUp is something we need to call anyways and allways (it is how the underlying NetMQ library works).